modules/local/simpleaf_quant.nf

```diff
@@ -23,25 +23,33 @@ process SALMON_ALEVIN {
     path  "versions.yml"                     , emit: versions
 
     script:
-    def args = task.ext.args ?: ''
+    def args   = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
 
     // separate forward from reverse pairs
     def (forward, reverse) = reads.collate(2).transpose()
     """
-    salmon alevin \\
-        -l ISR \\
-        -p $task.cpus \\
-        -1 ${forward.join( " " )} \\
-        -2 ${reverse.join( " " )} \\
-        --${protocol} \\
-        -i $index \\
-        --tgMap $txp2gene \\
-        --dumpFeatures --dumpMtx \\
-        $args \\
-        -o ${prefix}_alevin_results
-
-    gzip -cdf ${whitelist} > ${prefix}_alevin_results/alevin/whitelist.txt
+    # export required var
+    export ALEVIN_FRY_HOME=.
+
+    # prep simpleaf
+    simpleaf set-paths
+
+    # run simpleaf quant
+    gzip -dcf $whitelist > whitelist.txt
```

Comment:
this would break if the input file is called `whitelist.txt` (cannot redirect to a file of the same name)

The most efficent way is to have simpleaf read from a subshell (but may not be supported by simpleaf): 
```
   ... \\
   -u <(gzip -dcf $whitelist) \\
```

Another possibility is to append the extension: 
```
gzip -dcf $whitelist > ${whitelist}.uncompressed.txt
```

Reply:
just tested, and the tool does not accept subshell.

I will add the suffix.